vault/identity_store_entities.go

```diff
@@ -0,0 +1,463 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fatih/structs"
+	memdb "github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/vault/helper/locksutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"github.com/mitchellh/mapstructure"
+)
+
+// entityPaths returns the API endpoints supported to operate on entities.
+// Following are the paths supported:
+// entity - To register a new entity
+// entity/id - To lookup, modify, delete and list entities based on ID
+// entity/merge/id - To merge entities based on ID
+func entityPaths(i *identityStore) []*framework.Path {
+	return []*framework.Path{
+		{
+			Pattern: "entity",
+			Fields: map[string]*framework.FieldSchema{
+				"name": {
+					Type:        framework.TypeString,
+					Description: "Name of the entity",
+				},
+				"metadata": {
+					Type:        framework.TypeMap,
```

Comment:
If this is purely a map[string]string internally I wonder if we should actually treat this as a string slice. (e.g. use TypeCommaStringSlice), so values are just set using key=value pairs, like env vars. That way we can support this with the CLI, which currently can't support maps.